StakingERC721/contracts/StakingERC721.sol

```diff
@@ -0,0 +1,371 @@
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {IBunzz} from "./interfaces/IBunzz.sol";
+import {IStakingERC721} from "./interfaces/IStakingERC721.sol";
+
+error NoTokenIds();
+error WaitToFinish();
+error NotTokenOwner();
+error NotEnoughBalance();
+error TooHighReward();
+error FailedToWithdrawStaking();
+
+/**
+ * Stakes tokens for a certain duration and gets rewards according to their
+ * staked shares
+ */
+contract StakingERC721 is
+  IBunzz,
+  IStakingERC721,
+  IERC721Receiver,
+  Ownable,
+  Pausable,
+  ReentrancyGuard
+{
+  using SafeERC20 for IERC20;
+
+  // Address to staking ERC721 token
+  IERC721 public stakingToken;
+  // Address to rewards ERC20 token
+  IERC20 public rewardsToken;
+  // Block timestamp of the end of reward session
+  uint256 public periodFinish;
+  // Reward rate
+  uint256 public rewardRate;
+  // Rewards duration in seconds
+  uint256 public rewardsDuration = 7 days;
+  // Last timestamp of funding in seconds
+  uint256 public lastUpdateTime;
+  // Accumulated rewards token
+  uint256 public rewardPerTokenStored;
+
+  // Mapping for accumulated rewards token as temporary store
+  mapping(address => uint256) public userRewardPerTokenPaid;
+  // Mapping for rewards amount, <user account, rewards amount>
+  mapping(address => uint256) public rewards;
+
+  // Total staked token amount
+  uint256 public totalSupply;
+  // Balances of staked token per user
+  mapping(address => uint256) public balances;
+  // Mapping of staker/depositor of the token
+  mapping(uint256 => address) public stakedAssets;
+
+  /* -------------------------------------------------------------------------- */
+  /*                                   Events                                   */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * Emitted when user stakes staking token
+   */
+  event Staked(address indexed account, uint256 amount, uint256[] tokenIds);
+  /**
+   * Emited when user unstakes his/her staked token
+   */
+  event Unstaked(address indexed account, uint256 amount, uint256[] tokenIds);
+  /**
+   * Emitted when user claim his/her rewards token
+   */
+  event Claimed(address indexed account, uint256 amount);
+  /**
+   * Emitted when owner funds rewards token and restart rewarding session
+   */
+  event Funded(uint256 amount);
+  /**
+   * Emitted when rewards duration has been updated
+   */
+  event RewardsDurationUpdated(uint256 duration);
+  /**
+   * Emitted when owner recovered tokens from this contract
+   */
+  event Recovered(address token, uint256 amount);
```

Comment:
Missing NatSpec
Add descriptions for the events
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html